internal/topic/core.go

```diff
@@ -91,6 +92,25 @@ func (c *Core) CreateDeadLetterTopic(ctx context.Context, model *Model) error {
 	return c.repo.Create(ctx, model)
 }
 
+// UpdateTopic implements topic updation
+func (c *Core) UpdateTopic(ctx context.Context, m *Model) error {
+	startTime := time.Now()
+	defer func() {
+		topicOperationTimeTaken.WithLabelValues(env, "UpdateTopic").Observe(time.Now().Sub(startTime).Seconds())
+	}()
+
+	// validate if the topic exists
+	ok, err := c.Exists(ctx, m.Key())
+	if err != nil {
+		return err
+	}
+	if !ok {
+		return merror.New(merror.NotFound, "topic not found")
+	}
+
+	return c.repo.Create(ctx, m)
```

Comment:
Create or Update ?

Reply:
Its the same thing. Internally consul `PUT` gets called.